daemon/main.rs

```diff
@@ -104,3 +105,38 @@ fn exit_with_error(error: Error) {
     log::error!("{}", error_string.trim_start_matches("error: "));
     std::process::exit(error_code);
 }
+
+fn setup_lassie_tempdir(lassie_temp_dir: &Path) -> Result<()> {
+    if !lassie_temp_dir.is_dir() {
+        log::debug!("Creating Lassie tempdir {:?}", lassie_temp_dir,);
+        fs::create_dir_all(lassie_temp_dir)?;
+        return Ok(());
+    }
+
+    log::debug!(
+        "Cleaning left-over files in Lassie tempdir {:?}",
+        lassie_temp_dir
+    );
+
+    let lassie_files = fs::read_dir(lassie_temp_dir).with_context(|| {
+        format!(
+            "cannot list files in Lassie's temp dir {:?}",
+            lassie_temp_dir
+        )
+    })?;
+
+    for ent in lassie_files {
+        match ent {
+            Err(err) => log::warn!("Cannot parse dir entry in Lassie tempdir: {:?}", err),
+            Ok(f) => {
+                let p = f.path();
+                log::trace!("Removing Lassie temp file {:?}", p);
```

Comment:
I assume lassie doesn't place directories in here?

Reply:
If so, could you add this constraint as a comment please?

Reply:
> I assume lassie doesn't place directories in here?

Yes, that's my assumption too.

> If so, could you add this constraint as a comment please?

Great idea!